airwallex/src/main/java/com/airwallex/android/view/PaymentMethodsViewModel.kt

```diff
@@ -429,6 +437,88 @@ internal class PaymentMethodsViewModel(
+    }
+
+    suspend fun loadSchemaFields(paymentMethodType: AvailablePaymentMethodType): SchemaData? {
+        // Return cached data if available
+        schemaDataCache[paymentMethodType]?.let { return it }
+
+        AirwallexLogger.info("PaymentMethodsViewModel loadSchemaFields, type = ${paymentMethodType.name}")
+        val paymentMethod = PaymentMethod.Builder()
+            .setType(paymentMethodType.name)
+            .build()
+        paymentMethod.type?.let { type ->
+            if (requireHandleSchemaFields(paymentMethodType)) { // Have required schema fields
+                AirwallexLogger.info("PaymentMethodsViewModel get more payment Info fields on one-off flow.")
+                // 1.Retrieve all required schema fields of the payment method
+                val typeInfo = retrievePaymentMethodTypeInfo(type).getOrElse { exception ->
+                    _paymentFlowStatus.value = PaymentFlowStatus.ErrorAlert(exception.localizedMessage ?: "")
+                    schemaDataCache[paymentMethodType] = SchemaData()
+                    return null
+                }
+                // Ad hoc. Aligned with BE that we do not show Enum types in UI, instead we pass fixed values when we have the field.
+                listOf(COUNTRY_CODE to session.countryCode, OS_TYPE to OS_NAME, FLOW to AirwallexPaymentRequestFlow.IN_APP.value)
+                    .forEach { (key, value) ->
+                        if (typeInfo.needHiddenEnumParam(transactionMode, key)) {
+                            additionalParams[key] = value
+                        }
+                    }
+                val fields = typeInfo.filterRequiredFields(transactionMode) ?: return null
+                AirwallexLogger.info("PaymentMethodsViewModel loadSchemaFields: filterRequiredFields = $fields")
+                // 2.If all fields are hidden, start checkout directly
+                if (fields.isEmpty()) {
+                    schemaDataCache[paymentMethodType] = SchemaData()
+                    return SchemaData()
+                }
+                val bankField = fields.find { field -> field.type == DynamicSchemaFieldType.BANKS }
+                AirwallexLogger.info("PaymentMethodsViewModel loadSchemaFields: bankField = $bankField")
+                if (bankField == null) {
+                    return SchemaData(
+                        fields = fields,
+                        paymentMethod = paymentMethod,
+                        typeInfo = typeInfo,
+                    ).also { schemaData ->
+                        schemaDataCache[paymentMethodType] = schemaData
+                    }
+                }
+                // 3.If the bank is needed, need to retrieve the bank list.
+                val banks = retrieveBanks(type).getOrElse { exception ->
+                    _paymentFlowStatus.value = PaymentFlowStatus.ErrorAlert(exception.localizedMessage ?: "")
+                    schemaDataCache[paymentMethodType] = SchemaData()
```

Comment:
This will cause the payment method's schema not to reload after it failed to get the schema for the first time. I think we should always allow schema to reload.